src/dos/drive_virtual.cpp

```diff
@@ -636,6 +602,23 @@ const char* Virtual_Drive::GetLabel()
 	return "DOSBOX";
 }
 
+bool Virtual_Drive::equal_name(vfile_block_t cur_file, const char* search_name)
+{
+	unsigned int onpos = cur_file->onpos;
+	return strcasecmp(search_name,
+	                  (std::string(onpos ? vfilenames[onpos].shortname +
+	                                               std::string(1, '\\')
+	                                     : "") +
+	                   cur_file->name)
+	                          .c_str()) == 0;
+}
+vfile_block_t Virtual_Drive::find_file_by_name(const char* search_name)
+{
+	return find_file_by_predicate(first_file,
+	                              [this, search_name](vfile_block_t cur_file) {
+		                              return equal_name(cur_file, search_name);
```

Comment:
`cur_file` was named to be relative to it's linked-list position; suggest calling it `vfile`.